php/Plugin.php

```diff
@@ -144,14 +154,26 @@ public function version() {
 	/**
 	 * Sync the plugin version with the asset version.
 	 *
+	 * @param string $file Path to the file for which to check modified time.
+	 *
 	 * @return string
 	 */
-	public function asset_version() {
+	public function asset_version( $file = false ) {
 		if ( $this->is_debug() || $this->is_script_debug() ) {
-			return time();
+			return (string) time();
+		}
+
+		$version = $this->version();
+
+		if ( empty( $version ) ) {
+			if ( ! empty( $file ) && file_exists( $file ) ) {
+				$version = (string) filemtime( $file );
+			} else {
+				$version = $GLOBALS['wp_version'];
```

Comment:
Could we pass the WP version string as a dependency to the `__constructor()` to avoid coupling this class with a global state.

Reply:
Not sure I see the value of adding the global check as there should always be a version in the file header. We control that and is expected. If it's not there then we should throw an error in the `version()` method.
```
if ( $this->is_debug() || $this->is_script_debug() ) {
    return (string) time();
}

if ( ! empty( $file ) && file_exists( $file ) ) {
    return (string) filemtime( $file );
}

return $this->version();
```